subprojects/version-control/src/integTest/groovy/org/gradle/vcs/internal/GitVersionSelectionIntegrationTest.groovy

```diff
@@ -68,7 +68,7 @@ class GitVersionSelectionIntegrationTest extends AbstractIntegrationSpec {
         fixture.prepare()
     }
 
-    def "selects and builds from HEAD for latest.integration selector"() {
+    def "selects and builds from master for latest.integration selector"() {
```

Comment:
I think we could also ditch "latest.integration" as a magic string entirely.  We never had a real use case for it. It was useful to just have something that worked initially.

It seems like mapping "latest.integration" to master or remote HEAD is really workflow specific.  Why would latest.integration mean "master" over a branch called "development" or even "integration"?  With a workflow like git-flow, master is stable.  With Gradle's workflow, it's "mostly" stable.  With other workflows, master might be unstable and another branch may be the "integration" branch.